File: ticket-backend/src/main/java/aics/domain/fuzzy/dtos/FuzzyProfileDto.java
package aics.domain.fuzzy.dtos;

import aics.domain.fuzzy.etities.FuzzyProfile;
import aics.domain.fuzzy.models.FuzzyProfileData;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;

@Data
@Accessors(chain = true)
@RequiredArgsConstructor
public class FuzzyProfileDto implements Serializable {
    private Long fuzzyProfileId;
    private String name;
    private FuzzyProfileData fuzzyProfileData;
    private boolean enableDebug;
    private boolean active;

    public static FuzzyProfileDto fromFuzzyProfile(FuzzyProfile fuzzyProfile) {
        if (fuzzyProfile == null) {
            return null;
        }
        return new FuzzyProfileDto()
                .setFuzzyProfileId(fuzzyProfile.getFuzzyProfileId())
                .setName(fuzzyProfile.getName())
                .setFuzzyProfileData(fuzzyProfile.getFuzzyProfileData())
                .setEnableDebug(fuzzyProfile.isEnableDebug())
                .setActive(fuzzyProfile.isActive());
    }
}

File: ticket-frontend/src/modules/fuzzy/dtos/fuzzy-profile-dto.ts
import { FuzzyProfileData } from "../models/fuzzy-profile-data";

export class FuzzyProfileDto {
    fuzzyProfileId: number | null;
    name: string;
    fuzzyProfileData: FuzzyProfileData;
    enableDebug: boolean;
    active: boolean;
    useFuzzyTopsis: boolean;

    constructor(obj: {
        fuzzyProfileId: number | null,
        name: string,
        fuzzyProfileData: FuzzyProfileData,
        enableDebug: boolean,
        active: boolean,
        useFuzzyTopsis: boolean,
    }) {
        this.fuzzyProfileId = obj.fuzzyProfileId;
        this.name = obj.name;
        this.fuzzyProfileData = obj.fuzzyProfileData;
        this.enableDebug = obj.enableDebug;
        this.active = obj.active;
        this.useFuzzyTopsis = obj.useFuzzyTopsis;
    }

    static fromObj(obj: any): FuzzyProfileDto {
        if (!obj) {
            throw new Error('obj cannot be null');
        }
        return new FuzzyProfileDto({
            fuzzyProfileId: obj.fuzzyProfileId,
            name: obj.name,
            fuzzyProfileData: FuzzyProfileData.fromObj(obj.fuzzyProfileData),
            enableDebug: obj.enableDebug,
            active: obj.active,
            useFuzzyTopsis: obj.useFuzzyTopsis,
        });
    }

    deepClone(): FuzzyProfileDto {
        return new FuzzyProfileDto({
            fuzzyProfileId: this.fuzzyProfileId,
            name: this.name,
            fuzzyProfileData: this.fuzzyProfileData.deepClone(),
            enableDebug: this.enableDebug,
            active: this.active,
            useFuzzyTopsis: this.useFuzzyTopsis,
        })
    }
}